File: ReactRedux/src/models/ComponentMetadata.ts
import { ComponentTypeEnum, initComponentTypeEnum } from "../enums/ComponentTypeEnum";
import { PropMetadata, initPropMetadata } from "./PropMetadata";
import { FormMetadata, initFormMetadata } from "./FormMetadata";
export interface ComponentMetadata {
  idComponentMetadata:number;
  name?:string;
  caption?:string;
  description?:string;
  type:ComponentTypeEnum;
  typeString:string;
  idModelPropMetadata:number;
  modelPropMetadata?:PropMetadata;
  props:PropMetadata[];
  modelProp:boolean;
  idFormMetadata:number;
  formMetadata?:FormMetadata;
}

export const initComponentMetadata = {
  idComponentMetadata: 0,
  type: initComponentTypeEnum,
  typeString: '',
  idModelPropMetadata: 0,
  props: [],
  modelProp: false,
  idFormMetadata: 0,

}

File: DataAccessLayer/Models/ComponentMetadata.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DataAccessLayer.Dto
{
    public class ComponentMetadata
    {

        public ComponentMetadata()
        {
          Props = new HashSet<PropMetadata>();

        }

        [Key]
        public int IdComponentMetadata { get; set; }
        public string? Name { get; set; }
        public string? Caption { get; set; }
        public string? Description { get; set; }
        public string Type { get; set; }

        public int? IdModelPropMetadata { get; set; }

        
        [ForeignKey("IdModelPropMetadata")]
        public virtual PropMetadata? ModelPropMetadata { get; set; }
        
        public virtual ICollection<PropMetadata>? Props { get; set; }        public bool ModelProp { get; set; }

        public int? IdFormMetadata { get; set; }

        
        [ForeignKey("IdFormMetadata")]
        public virtual FormMetadata? FormMetadata { get; set; }
    }
}
